src/DotNetWorker.Core/Context/DefaultFunctionContext.cs

```diff
@@ -58,11 +58,11 @@ public override IServiceProvider InstanceServices
 
         public override RetryContext RetryContext => Features.GetRequired<IExecutionRetryFeature>().Context;
 
-        public void Dispose()
+        public async ValueTask DisposeAsync()
         {
-            if (_instanceServicesScope != null)
+            if (_instanceServicesScope is IAsyncDisposable asyncServiceScope)
```

Comment:
explicit null check is not needed as the "is" expression will handle that as well.